### PR TITLE
Front-End: added Board Of Appeals letter support to CST DDL

### PR DESCRIPTION
**Pull request by samcoforma**
## Summary

- *added Board Of Appeals letter support to CST DDL and created feature flag for future use*

## Related issue(s)
- Related PR: [department-of-veterans-affairs/va.gov-team#0000](https://github.com/department-of-veterans-affairs/vets-api/pull/12329)
- *Original ticket: https://github.com/department-of-veterans-affairs/va.gov-team/issues/55272*

## Testing done

- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected*
- *Describe the tests completed and the results*
- *Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)*

## Screenshots
_Note: This field is mandatory for component work and UI changes (non-component work should NOT have screenshots)._

| | Before | After |
| --- | --- | --- |
| Mobile | | |
| Desktop | | |

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature
- [ ]  [Accessibility foundational testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/claims-status/selectors/index.js [hunk 1]
```diff
@@ -13,5 +13,9 @@ export const showClaimLettersFeature = state =>
 export const cstUseLighthouse = state =>
   toggleValues(state)[FEATURE_FLAG_NAMES.cstUseLighthouse];
 
+// 'cst_include_ddl_boa_letters'
+export const cstIncludeDdlBoaLetters = state =>
+  toggleValues(state)[FEATURE_FLAG_NAMES.cstIncludeDdlBoaLetters];
```
Comment (kjduensing) [inline src/applications/claims-status/selectors/index.js:18]: Why do we need this change and the change in `featureFlagNames.js`?

Comment (samcoforma) [reply src/applications/claims-status/selectors/index.js:18]: @kjduensing We're going to use it to toggle a link from the Appeals listing page soon... cc: @jerekshoe

Comment (kjduensing) [reply src/applications/claims-status/selectors/index.js:18]: Ok thanks. You're saying it's not relevant to this PR, but we'll need it in the future?

Comment (samcoforma) [reply src/applications/claims-status/selectors/index.js:18]: Yeah, basically. I can remove it if it seems like a bad idea to push it with this PR @kjduensing

Comment (kjduensing) [reply src/applications/claims-status/selectors/index.js:18]: Cool beans, thanks for clarifying. I dunno that it's a bad idea, it just muddies the purpose of this PR a little. Up to you, mostly just trying to understand.

**Review by kjduensing — APPROVED** [review]
LGTM 🦅

**Review by harshilmoradia — APPROVED** [review]

**Pull request merged**